plugins/com.google.cloud.tools.eclipse.ui.util/src/com/google/cloud/tools/eclipse/ui/util/MessageConsoleUtilities.java

```diff
@@ -61,4 +61,31 @@ public static MessageConsole getMessageConsole(String consoleName,
     }
     return messageConsole;
   }
+  
+  public static <C extends MessageConsole> C findOrCreateConsole(String name, ConsoleFactory<C> factory) {
```

Comment:
This method is actually not used by deploy, only the for local App Engine runs. I did not want to change that to use console type in this PR, just renamed the method to describe better what it does.